### ISSUE TITLE
Matching parens with global-hl-line-mode enabled

### ISSUE THREAD

**Issue by mwlodarczak** [tmalsburg/tango-plus-theme#6]
Hi!

When global-hl-line-mode is enabled, the current line gets the same highlight colour as the matching paren, making it invisible.

Kind regards,
Marcin

**Comment by tmalsburg** [comment]
Yes, this has bothered me for a long time, but I'm not sure how to deal with this.  Suggestions?

**Comment by mwlodarczak** [comment]
On 05/07/2017 06:25 PM, Titus von der Malsburg wrote:
> Yes, this has bothered me for a long time, but I'm not 
> sure how to deal with this.  Suggestions?
> 

Why not set `show-paren-match' to something contrastive
enough? red-2 seems to work fine, as far as I am concerned.

**Comment by tmalsburg** [comment]
Not sure, red is used for errors, stuff that is going to be deleted, and similar things.  Also, you seem to be using `parenthesis` as your `show-paren-style`, so only the parenthesis would be red, but other people are using `expression` where everything between the parentheses would also be red.  Seems a bit too much.  My personal resolution for his problem was to disable hl-line-mode for lisp code.

**Comment by aminb** [comment]
Chiming in to say this had bothered me as well. I tired a few different colours for `show-paren-match` and ended up going with `butter-1`. I've been using it for a few months and it works for me.

```diff
-   `(show-paren-match               ((,class (:inherit highlight))))
+   `(show-paren-match               ((,class (:background ,butter-1))))
```

**Comment by tmalsburg** [comment]
Hi @aminb, thanks for the suggestion.  The problem with using butter-1 is that it is indistinguishable from marked regions and this could be confusing.  Would it be okay to use different shades of grey for highlighted lines and matching parentheses?

**Comment by aminb** [comment]
Hi @tmalsburg, you have good point: when marking regions, you wouldn't be able to distinguish the highlighted matching paren, but I to be honest I haven't had any issues with that so far.

Anyways; I just tried your suggestion of using a different shade of grey for `show-paren-match`. Specifically, I tried `alum-3`; it looks quite alright to me.

**Comment by tmalsburg** [comment]
I guess what we really need is a way to modify the theme depending on the value of `show-paren-style` but I'm not sure whether the theming infrastructure has support for this.  We can of course introduce a conditional in the definition of the theme but this will not respond to changed of `show-paren-style` after the theme is loaded.
